dNR: Implement the optional filter parameter of getDynamicRules and getSessionRules

elijahsawyers opened the pull request:
#### d6b240ab340c771414e1243bdc8da362e941451f
<pre>
dNR: Implement the optional filter parameter of getDynamicRules and getSessionRules
<a href="https://bugs.webkit.org/show_bug.cgi?id=290589">https://bugs.webkit.org/show_bug.cgi?id=290589</a>
<a href="https://rdar.apple.com/147782333">rdar://147782333</a>

Reviewed by Timothy Hatcher.

This patch implements the filter parameter of the getDynamicRules and the
getSessionRules dNR methods.

This parameter is an object that has an optional ruleIds array of numbers,
which defines which rules to return.

Wrote new tests to validate this change.

* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
(WebKit::WebExtensionContext::declarativeNetRequestGetDynamicRules):
(WebKit::WebExtensionContext::declarativeNetRequestGetSessionRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::loadDeclarativeNetRequestRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.h:
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.mm:
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore getRulesWithRuleIDs:completionHandler:]):
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore _getRulesWithRuleIDs:errorMessage:]):
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore getRulesWithCompletionHandler:]): Deleted.
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore _getRulesWithOutErrorMessage:]): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.messages.in:
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
(WebKit::WebExtensionAPIDeclarativeNetRequest::getDynamicRules):
(WebKit::WebExtensionAPIDeclarativeNetRequest::getSessionRules):
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIDeclarativeNetRequest.h:
* Source/WebKit/WebProcess/Extensions/Interfaces/WebExtensionAPIDeclarativeNetRequest.idl:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, GetSessionRules)):
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, GetDynamicRules)):

Canonical link: <a href="https://commits.webkit.org/292892@main">https://commits.webkit.org/292892@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f67a32b0db3fb473430be0e13e1e4d1639516ca2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97330 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16955 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7169 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102416 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47858 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99375 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17248 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25404 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74147 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31338 "Found 1 new test failure: workers/worker-to-worker.html (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100333 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13044 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88019 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54488 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12803 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5901 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47301 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82840 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5983 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104436 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24407 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17794 "Found 2 new test failures: imported/w3c/web-platform-tests/workers/abrupt-completion.html ipc/send-gpu-GetShareableBitmap-RemoteRenderingBackend.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83191 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24780 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84144 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82610 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27160 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4843 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17953 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15729 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24371 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29538 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24193 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27507 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25767 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
```diff
@@ -4858,7 +4858,7 @@ void effectiveAppearanceDidChange(Inspector::ExtensionAppearance appearance) ove
 
     auto addDynamicAndStaticRules = [this, protectedThis = Ref { *this }, addStaticRulesets = WTFMove(addStaticRulesets), allJSONData = RetainPtr { allJSONData }] () mutable {
-        [declarativeNetRequestDynamicRulesStore() getRulesWithCompletionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addStaticRulesets = WTFMove(addStaticRulesets), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
+        [declarativeNetRequestDynamicRulesStore() getRulesWithRuleIDs:@[] completionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addStaticRulesets = WTFMove(addStaticRulesets), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
```
```suggestion
        [declarativeNetRequestDynamicRulesStore() getRulesWithRuleIDs:@[ ] completionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addStaticRulesets = WTFMove(addStaticRulesets), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
```

elijahsawyers replied:
The linter complains about this.

xeenon replied:
The linter is wrong and can be ignored. We should fix it someday.

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
```diff
@@ -4882,7 +4882,7 @@ void effectiveAppearanceDidChange(Inspector::ExtensionAppearance appearance) ove
     };
 
-    [declarativeNetRequestSessionRulesStore() getRulesWithCompletionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addDynamicAndStaticRules = WTFMove(addDynamicAndStaticRules), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
+    [declarativeNetRequestSessionRulesStore() getRulesWithRuleIDs:@[] completionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addDynamicAndStaticRules = WTFMove(addDynamicAndStaticRules), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
```
```suggestion
    [declarativeNetRequestSessionRulesStore() getRulesWithRuleIDs:@[ ] completionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, addDynamicAndStaticRules = WTFMove(addDynamicAndStaticRules), allJSONData = RetainPtr { allJSONData }](NSArray *rules, NSString *errorMessage) mutable {
```

elijahsawyers replied:
Ditto: https://github.com/WebKit/WebKit/pull/43154#discussion_r2019463960

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -61,6 +61,9 @@
 static NSString * const getMatchedRulesMinTimeStampKey = @"minTimeStamp";
 
+static NSString * const getDynamicRulesRuleIDsKey = @"ruleIds";
+static NSString * const getSessionRulesRuleIDsKey = @"ruleIds";
```
You can just have one static for this since it is the same value.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -138,9 +141,31 @@
+    NSMutableSet *otherFilterKeys = [NSMutableSet setWithArray:filter.allKeys];
+    [otherFilterKeys minusSet:[NSSet setWithObject:getDynamicRulesRuleIDsKey]];
+    if (otherFilterKeys.count) {
+        callback->reportError(toErrorString(@"declarativeNetRequest.getDynamicRules()", nullString(), [NSString stringWithFormat:@"unexpected property: '%@' in 'filter' parameter", otherFilterKeys.allObjects.firstObject]));
```
```suggestion
        callback->reportError(toErrorString(@"declarativeNetRequest.getDynamicRules()", "filter"_s, [NSString stringWithFormat:@"unexpected property '%@'", otherFilterKeys.allObjects.firstObject]));
```

xeenon replied:
We don't usually error on unexpected properties in Safari, so I would just remove this altogether. I'm pushing all other browsers to do similar in the WECG.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -138,9 +141,31 @@
+        return;
+    }
+
+    Vector<int> ruleIDs;
```
Is `int` the right type? What about overflows? `ssize_t` might be best.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -138,9 +141,31 @@
+
+    Vector<int> ruleIDs;
+    for (NSNumber *ruleID in objectForKey<NSArray>(filter, getDynamicRulesRuleIDsKey, false, NSNumber.class))
+        ruleIDs.append(ruleID.integerValue);
```
There is a CocoaVector.h template helper for this, I just don't remember what it is and I am not at my computer. Consider using it instead of writing a loop yourself.

elijahsawyers replied:
Which one did you have in mind? I couldn't find it.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -194,9 +219,31 @@
+    if (otherFilterKeys.count) {
+        callback->reportError(toErrorString(@"declarativeNetRequest.getSessionRules()", nullString(), [NSString stringWithFormat:@"unexpected property: '%@' in 'filter' parameter", otherFilterKeys.allObjects.firstObject]));
+        return;
+    }
```
Ditto.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -194,9 +219,31 @@
+
+    Vector<int> ruleIDs;
+    for (NSNumber *ruleID in objectForKey<NSArray>(filter, getSessionRulesRuleIDsKey, false, NSNumber.class))
+        ruleIDs.append(ruleID.integerValue);
```
Ditto.

elijahsawyers replied:
https://github.com/WebKit/WebKit/pull/43154#discussion_r2019533707

xeenon commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
```diff
@@ -520,6 +520,70 @@
+        @"await browser.declarativeNetRequest.updateSessionRules({ addRules: [{ id: 2, priority: 1, action: {type: 'block'}, condition: { urlFilter: 'bar' } }] })",
+        @"await browser.declarativeNetRequest.updateSessionRules({ addRules: [{ id: 3, priority: 1, action: {type: 'block'}, condition: { urlFilter: 'baz' } }] })",
+
+        @"browser.test.assertRejects(browser.declarativeNetRequest.getSessionRules({ foo: [ 1 ] }))",
```
This shouldn't be an error.

xeenon commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
```diff
@@ -586,6 +650,70 @@
+        @"await browser.declarativeNetRequest.updateDynamicRules({ addRules: [{ id: 2, priority: 1, action: {type: 'block'}, condition: { urlFilter: 'bar' } }] })",
+        @"await browser.declarativeNetRequest.updateDynamicRules({ addRules: [{ id: 3, priority: 1, action: {type: 'block'}, condition: { urlFilter: 'baz' } }] })",
+
+        @"browser.test.assertRejects(browser.declarativeNetRequest.getDynamicRules({ foo: [ 1 ] }))",
```
Ditto.

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -138,9 +140,24 @@
+
+    Vector<size_t> ruleIDs;
+    for (NSNumber *ruleID in objectForKey<NSArray>(filter, getDynamicOrSessionRulesRuleIDsKey, false, NSNumber.class))
+        ruleIDs.append(ruleID.unsignedIntValue);
```
```suggestion
    auto ruleIDs = makeVector<double>(filter[getDynamicOrSessionRulesRuleIDsKey], [](NSNumber *ruleID) { return ruleID.doubleValue; });
```

elijahsawyers replied:
Getting an error for that:

`note: candidate template ignored: substitution failure [with MapFunctionType = double]: no type named 'type' in 'std::invoke_result<double, id>'`

xeenon commented on Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -194,9 +211,24 @@
+
+    Vector<size_t> ruleIDs;
+    for (NSNumber *ruleID in objectForKey<NSArray>(filter, getDynamicOrSessionRulesRuleIDsKey, false, NSNumber.class))
+        ruleIDs.append(ruleID.unsignedIntValue);
```
```suggestion
    auto ruleIDs = makeVector<double>(filter[getDynamicOrSessionRulesRuleIDsKey], [](NSNumber *ruleID) { return ruleID.doubleValue; });
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
```diff
@@ -795,9 +795,9 @@ class WebExtensionContext : public API::ObjectImpl<API::Object::Type::WebExtensi
     void declarativeNetRequestToggleRulesets(const Vector<String>& rulesetIdentifiers, bool newValue, NSMutableDictionary *rulesetIdentifiersToEnabledState);
     void declarativeNetRequestGetMatchedRules(std::optional<WebExtensionTabIdentifier>, std::optional<WallTime> minTimeStamp, CompletionHandler<void(Expected<Vector<WebExtensionMatchedRuleParameters>, WebExtensionError>&&)>&&);
-    void declarativeNetRequestGetDynamicRules(CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
+    void declarativeNetRequestGetDynamicRules(Vector<size_t>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
```
```suggestion
    void declarativeNetRequestGetDynamicRules(Vector<double>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
```diff
@@ -795,9 +795,9 @@ class WebExtensionContext : public API::ObjectImpl<API::Object::Type::WebExtensi
+    void declarativeNetRequestGetDynamicRules(Vector<size_t>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
     void declarativeNetRequestUpdateDynamicRules(String&& rulesToAddJSON, Vector<double>&& ruleIDsToDelete, CompletionHandler<void(Expected<void, WebExtensionError>&&)>&&);
-    void declarativeNetRequestGetSessionRules(CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
+    void declarativeNetRequestGetSessionRules(Vector<size_t>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
```
```suggestion
    void declarativeNetRequestGetSessionRules(Vector<double>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&&);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.mm:
```diff
@@ -191,25 +191,42 @@ - (void)getRulesWithCompletionHandler:(void (^)(NSArray<NSDictionary<NSString *,
+        _WKWebExtensionSQLiteStatement *statement = [[_WKWebExtensionSQLiteStatement alloc] initWithDatabase:_database query:query];
+
+        for (NSUInteger i = 0; i < ruleIDs.count; i++)
+            [statement bindInt:ruleIDs[i].intValue atParameterIndex:i + 1];
```
```suggestion
            [statement bindInt64:ruleIDs[i].integerValue atParameterIndex:i + 1];
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -350,9 +350,13 @@
+void WebExtensionContext::declarativeNetRequestGetDynamicRules(Vector<size_t>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&& completionHandler)
 {
-    [declarativeNetRequestDynamicRulesStore() getRulesWithCompletionHandler:makeBlockPtr([protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)](NSArray *rules, NSString *errorMessage) mutable {
+    auto *ruleIDs = createNSArray(filter, [this](int ruleID) -> NSNumber * {
```
```suggestion
    auto *ruleIDs = createNSArray(filter, [this](auto ruleID) -> NSNumber * {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
```diff
@@ -388,9 +392,13 @@
+void WebExtensionContext::declarativeNetRequestGetSessionRules(Vector<size_t>&& filter, CompletionHandler<void(Expected<String, WebExtensionError>&&)>&& completionHandler)
 {
-    [declarativeNetRequestSessionRulesStore() getRulesWithCompletionHandler:makeBlockPtr([protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)](NSArray *rules, NSString *errorMessage) mutable {
+    auto *ruleIDs = createNSArray(filter, [this](int ruleID) -> NSNumber * {
```
```suggestion
    auto *ruleIDs = createNSArray(filter, [this](auto ruleID) -> NSNumber * {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.mm:
```diff
@@ -191,25 +191,42 @@ - (void)getRulesWithCompletionHandler:(void (^)(NSArray<NSDictionary<NSString *,
+
+        NSString *joinedBindings = [bindings componentsJoinedByString:@", "];
+        NSString *query = [NSString stringWithFormat:@"SELECT * FROM %@ WHERE id IN (%@)", _tableName, joinedBindings];
+        _WKWebExtensionSQLiteStatement *statement = [[_WKWebExtensionSQLiteStatement alloc] initWithDatabase:_database query:query];
```
```suggestion
        auto *joinedBindings = [bindings componentsJoinedByString:@", "];
        auto *query = [NSString stringWithFormat:@"SELECT * FROM %@ WHERE id IN (%@)", _tableName, joinedBindings];
        auto *statement = [[_WKWebExtensionSQLiteStatement alloc] initWithDatabase:_database query:query];
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.mm:
```diff
@@ -191,25 +191,42 @@ - (void)getRulesWithCompletionHandler:(void (^)(NSArray<NSDictionary<NSString *,
+    _WKWebExtensionSQLiteRowEnumerator *rows;
+
+    if (ruleIDs.count) {
+        NSMutableArray<NSString *> *bindings = [NSMutableArray array];
```
```suggestion
        auto *bindings = [NSMutableArray array];
```

elijahsawyers committed:
dNR: Implement the optional filter parameter of getDynamicRules and getSessionRules
https://bugs.webkit.org/show_bug.cgi?id=290589
rdar://147782333

Reviewed by Timothy Hatcher.

This patch implements the filter parameter of the getDynamicRules and the
getSessionRules dNR methods.

This parameter is an object that has an optional ruleIds array of numbers,
which defines which rules to return.

Wrote new tests to validate this change.

* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
(WebKit::WebExtensionContext::declarativeNetRequestGetDynamicRules):
(WebKit::WebExtensionContext::declarativeNetRequestGetSessionRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::loadDeclarativeNetRequestRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.h:
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestSQLiteStore.mm:
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore getRulesWithRuleIDs:completionHandler:]):
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore _getRulesWithRuleIDs:errorMessage:]):
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore getRulesWithCompletionHandler:]): Deleted.
(-[_WKWebExtensionDeclarativeNetRequestSQLiteStore _getRulesWithOutErrorMessage:]): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.messages.in:
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm:
(WebKit::WebExtensionAPIDeclarativeNetRequest::getDynamicRules):
(WebKit::WebExtensionAPIDeclarativeNetRequest::getSessionRules):
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIDeclarativeNetRequest.h:
* Source/WebKit/WebProcess/Extensions/Interfaces/WebExtensionAPIDeclarativeNetRequest.idl:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, GetSessionRules)):
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, GetDynamicRules)):

Canonical link: https://commits.webkit.org/292892@main
